starknet-rpc-test/src/lib.rs

```diff
@@ -141,28 +148,58 @@ fn get_free_port() -> Result<u16, TestError> {
 }
 
 impl MadaraClient {
-    async fn init(execution: ExecutionStrategy) -> Result<Self, TestError> {
+    async fn init(
+        execution: ExecutionStrategy,
+        settlement: Option<Settlement>,
+        madara_path: Option<PathBuf>,
+        madara_log: bool,
+    ) -> Result<Self, TestError> {
         let free_port = get_free_port()?;
 
         let manifest_path = Path::new(&env!("CARGO_MANIFEST_DIR"));
         let repository_root = manifest_path.parent().expect("Failed to get parent directory of CARGO_MANIFEST_DIR");
 
         std::env::set_current_dir(repository_root).expect("Failed to change working directory");
 
+        let settlement_arg = match settlement {
+            Some(arg) => format!("--settlement={arg}"),
+            None => String::new(),
+        };
+
+        let madara_path_arg = match madara_path {
+            Some(arg) => format!("--madara-path={}", arg.display()),
+            None => String::new(),
+        };
+
+        let (stdout, stderr) = match madara_log {
+            true => {
+                let logs_dir = Path::join(repository_root, Path::new("logs"));
+                create_dir_all(logs_dir.clone()).expect("Failed to create logs dir");
+
+                (
+                    Stdio::from(File::create(Path::join(&logs_dir, Path::new("madara-stdout-log.txt"))).unwrap()),
+                    Stdio::from(File::create(Path::join(&logs_dir, Path::new("madara-stderr-log.txt"))).unwrap()),
+                )
+            }
+            false => (Stdio::null(), Stdio::null()),
+        };
+
         let child_handle = Command::new("cargo")
-		// Silence Madara stdout and stderr
-		.stdout(Stdio::null())
-		.stderr(Stdio::null())
-		.args([
-			"run",
-			"--release",
-			"--",
-			"--alice",
-			"--sealing=manual",
-			&format!("--execution={execution}"),
-			"--chain=dev",
-			"--tmp",
-			&format!("--rpc-port={free_port}")
+            // Silence Madara stdout and stderr
+            .stdout(stdout)
+            .stderr(stderr)
```

Comment:
Silence them back before merging

Reply:
I've added a flag to `MadaraClient::init` to enable/disable logging, I suppose this might be useful for other scenarios as well. I will set it to false everywhere, also can make it feature enabled if you'd like

Reply:
I made logs optional, now they can be enabled by providing MADARA_LOG env variable which should contain a relative log folder, is that ok? Generally I think there might be other cases when you'd want to check Madara logs for troubleshooting (at least I couldn't debug efficiently without them)

Reply:
Added readme for integration tests